client/era/era_service.go

```diff
@@ -0,0 +1,158 @@
+}
+
+func (sc ServiceClient) GetProfiles(ctx context.Context, engine string, profileType string, id string, name string) (*ListProfileResponse, error) {
+	var httpReq *http.Request
+	var err error
+	path := makePathProfiles(engine, profileType, id, name)
+
+	httpReq, err = sc.c.NewRequest(ctx, http.MethodGet, path, nil)
+
+	if err != nil {
+		return nil, err
+	}
+	res := new(ListProfileResponse)
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func (sc ServiceClient) GetCluster(ctx context.Context, id string, name string) (*ListClusterResponse, error) {
+	var path string
+	if id != "" {
+		path = fmt.Sprintf("/clusters/%s", id)
+	}
+	if name != "" {
+		path = fmt.Sprintf("/clusters/name/%s", name)
+	}
+	httpReq, err := sc.c.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, err
+	}
+	res := new(ListClusterResponse)
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func (sc ServiceClient) ListClusters(ctx context.Context) (*ClusterListResponse, error) {
+	httpReq, err := sc.c.NewRequest(ctx, http.MethodGet, "/clusters", nil)
+	if err != nil {
+		return nil, err
+	}
+	res := new(ClusterListResponse)
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func (sc ServiceClient) GetSLA(ctx context.Context, id string, name string) (*ListSLAResponse, error) {
+	var path string
+	if id != "" {
+		path = fmt.Sprintf("/slas/%s", id)
+	}
+	if name != "" {
+		path = fmt.Sprintf("/slas/name/%s", name)
+	}
+	httpReq, err := sc.c.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, err
+	}
+	res := new(ListSLAResponse)
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func (sc ServiceClient) ListSLA(ctx context.Context) (*SLAResponse, error) {
+	httpReq, err := sc.c.NewRequest(ctx, http.MethodGet, "/slas", nil)
+	if err != nil {
+		return nil, err
+	}
+	res := new(SLAResponse)
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func makeListProfilePath(engine string, profileType string) string {
+	if engine != "" && profileType != "" {
+		return fmt.Sprintf("/profiles?engine=%s&type=%s", engine, profileType)
+	}
+	if engine != "" {
+		return fmt.Sprintf("/profiles?engine=%s", engine)
+	} else if profileType != "" {
+		return fmt.Sprintf("/profiles?type=%s", profileType)
+	}
+	return "/profiles"
+}
+
+func makePathProfiles(engine string, ptype string, id string, name string) string {
+	if engine != "" {
+		path := "/profiles?engine=" + engine
+		if ptype != "" {
+			path = path + "&type=" + ptype
+		}
+		if id != "" {
+			path = path + "&id=" + id
+		}
+		if name != "" {
+			path = path + "&name=" + name
+		}
+		return path
+	}
+	if ptype != "" {
+		path := "/profiles?type=" + ptype
+		if id != "" {
+			path = path + "&id=" + id
+		}
+		if name != "" {
+			path = path + "&name=" + name
+		}
+		return path
+	}
+
+	if id != "" {
+		path := "/profiles?id=" + id
+		if name != "" {
+			path = path + "&name=" + name
+		}
+		return path
+	}
+
+	if name != "" {
+		path := "/profiles?name=" + name
+		return path
+	}
+	return ""
+}
```

Comment:
In this file code at multiple places there are continuous multiple "if" clauses. Can we use "else-if" to avoid edge cases to be at safer side

Reply:
each "if" statement has a return path.